ROADMAP.md update for stages and forward view

qnetter opened the pull request:
## What this PR does / why we need it

This PR updates the published roadmap to include development of features and a more extended forward view.
## Details for the Release Notes (PLEASE PROVIDE)


```release-note
NONE
```

## Which issue(s) this PR fixes
<!--
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
-->
Fixes: #

## Describe testing done for PR

Validation by core engineering and other stakeholders

## Special notes for your reviewer
<!--
Add any things that reviewers should be aware of as they review
your PR.

Example: Please verify how I handled foo aligns with overall plan.
-->

qnetter committed:
ROADMAP.md update for state and forward view

kcoriordan commented on ROADMAP.md:
```diff
@@ -1,61 +1,48 @@
+||Validate and improve end-to-end tests across all supported infrastructure providers.|Awaiting proposal|TBD||
+|Rich package library|Work with community to solicit and prioritize desired packages||0-3 months|
+||Pursue community and ISV package lifecycle and configuration integrations via Carvel and Tanzu Framework|n/aongoing||
+|Positioning Community |dition as leading-edge upstream for Tanzu products|Community Edition produces its own Bill of Materials (BOM), Tanzu Kubernetes Release (TKr), OCI container images, host images,and packages for CLI plugins and in-cluster software|Awaiting Proposals|TBD||
```
```suggestion
|Positioning Community Edition as leading-edge upstream for Tanzu products|Community Edition produces its own Bill of Materials (BOM), Tanzu Kubernetes Release (TKr), OCI container images, host images,and packages for CLI plugins and in-cluster software|Awaiting Proposals|TBD||
```

qnetter replied:
Edited.

kcoriordan commented on ROADMAP.md:
```diff
@@ -1,61 +1,48 @@
+|Rich package library|Work with community to solicit and prioritize desired packages||0-3 months|
+||Pursue community and ISV package lifecycle and configuration integrations via Carvel and Tanzu Framework|n/aongoing||
+|Positioning Community |dition as leading-edge upstream for Tanzu products|Community Edition produces its own Bill of Materials (BOM), Tanzu Kubernetes Release (TKr), OCI container images, host images,and packages for CLI plugins and in-cluster software|Awaiting Proposals|TBD||
+|Application Platform|Installable package is made available containing pre-configured, minimally-viable, platform services that enable developers to run their software (contour, kpack, cert-manager, ?, ?)|Awaiting proposal|Implementation in Progress Proposal: (TODO: tracker proposal for all components)| Beginning in v0.10; please see [add issue once available]|0-3 monthsand 3-6 months|
```
```suggestion
|Application Platform|Installable package is made available containing pre-configured, minimally-viable, platform services that enable developers to run their software (contour, kpack, cert-manager, ?, ?)|Awaiting proposal|Implementation in Progress Proposal: (TODO: tracker proposal for all components)| Beginning in v0.10; please see [add issue once available]|0-3 months and 3-6 months|
```

kcoriordan commented on ROADMAP.md:
```diff
@@ -1,61 +1,48 @@
+|Robust Kubernetes platform|Limited-internet (including airgapped) support|Awaiting Proposal|TBD||
+||Validate and improve end-to-end tests across all supported infrastructure providers.|Awaiting proposal|TBD||
+|Rich package library|Work with community to solicit and prioritize desired packages||0-3 months|
+||Pursue community and ISV package lifecycle and configuration integrations via Carvel and Tanzu Framework|n/aongoing||
```
```suggestion
||Pursue community and ISV package lifecycle and configuration integrations via Carvel and Tanzu Framework|ongoing||
```

kcoriordan commented on ROADMAP.md:
```diff
@@ -1,61 +1,48 @@
+||Validate and improve end-to-end tests across all supported infrastructure providers.|Awaiting proposal|TBD||
+|Rich package library|Work with community to solicit and prioritize desired packages||0-3 months|
+||Pursue community and ISV package lifecycle and configuration integrations via Carvel and Tanzu Framework|n/aongoing||
+|Positioning Community |dition as leading-edge upstream for Tanzu products|Community Edition produces its own Bill of Materials (BOM), Tanzu Kubernetes Release (TKr), OCI container images, host images,and packages for CLI plugins and in-cluster software|Awaiting Proposals|TBD||
```
```suggestion
|Positioning Community |dition as leading-edge upstream for Tanzu products|Community Edition produces its own Bill of Materials (BOM), Tanzu Kubernetes Release (TKr), OCI container images, host images, and packages for CLI plugins and in-cluster software|Awaiting Proposals|TBD||
```

kcoriordan commented on ROADMAP.md:
```diff
@@ -1,61 +1,48 @@
+
+This document provides an overview of the major themes driving Tanzu Community Edition development, as well as constituent features and capabilities. Roadmap items are tracked by their implementation state within the enhancements proposal process: Awaiting Proposal, Proposal in Review, and Implementation in Progress. Most items are gathered from the community or include a feedback loop with the c ...
+
+This roadmap should serve as a reference point for Tanzu Community Edition users to help us prioritize existing features and provide input on unmet needs, and for contributors to understand where the project is heading. A roadmap also helps our contributor community to determine if a proposed enhancement conflicts with the long term plan.
```
```suggestion
This roadmap should serve as a reference point for Tanzu Community Edition users to help us prioritize existing features and provide input on unmet needs, and for contributors to understand where the project is heading. A roadmap also helps our contributor community to determine if a proposed enhancement conflicts with the long-term plan.
```

kcoriordan reviewed (commented):
Very minor edits - plus the table is broken - let me know if you want me to fix it  up

joshrosso reviewed (changes requested):
blocking for markdown lint & url lint issues.

@qnetter let us know if we can help with running those lint checks locally.

Will re-review once resolved.

qnetter committed:
Update ROADMAP.md

Mdlint fixes

qnetter commented:
@joshrosso the ruling fail, works fine for me manually - any ideas?

qnetter committed:
Update ROADMAP.md

Co-authored-by: Kay ORiordan <koriordan@vmware.com>

qnetter committed:
Update ROADMAP.md

Co-authored-by: Kay ORiordan <koriordan@vmware.com>

qnetter committed:
Update ROADMAP.md

qnetter committed:
Update ROADMAP.md

Co-authored-by: Kay ORiordan <koriordan@vmware.com>

joshrosso commented:
The table is not formatted correctly:

![image](https://user-images.githubusercontent.com/6200057/151429376-530defb2-23c2-4a99-b641-ac9daa8cdb40.png)

joshrosso commented:
@qnetter the engineering team is waiting on this roadmap to determine where we should go from here.

Do you think we could knock out the formatting asap? Or would you like us to step in and fix the table?

qnetter committed:
Update ROADMAP.md

Fix table.

qnetter committed:
Update ROADMAP.md

Last few typos.

stmcginnis commented on ROADMAP.md:
```diff
@@ -2,60 +2,47 @@
+
+Last Updated:  January 2022
+
+| Theme | Feature | Status (Phase) | Targeted Release |
```
You're missing the separator line after the header columns.

https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/organizing-information-with-tables#creating-a-table

stmcginnis commented on ROADMAP.md:
```diff
@@ -2,60 +2,47 @@
+
+| Theme | Feature | Status (Phase) | Targeted Release |
+| Build user community | Reward community support | n/a | ongoing |
+| |Community roadmap prioritization| | n/a | ongoing |
```
There are five columns defined here, but only four for the table.

stmcginnis commented on ROADMAP.md:
```diff
@@ -2,60 +2,47 @@
+| |Community roadmap prioritization| | n/a | ongoing |
+| Enhance cloud-native user experience for skilled and new users | Unmanaged-cluster model introduced to enable minimal-cluster deployments in under 4 minutes that can run on consumer hardware. | Implementation in Progress | v0.11 |
+| | Work to reduce time to cluster availability on vSphere and public clouds (Cluster API v1.1
+Proposal) | Awaiting Proposal | TBD |
```
Line breaks in the middle of a row aren't handled well by the markdown parsers.

stmcginnis reviewed (commented):
A few issues pointed out.

joshrosso commented:
> source now looks good but formatted version does not reflect changes.

The formatted version does reflect the source, but the source has issue.

I'm going to update the formatting and push a new commit.

Commit:
fix table formatting

Signed-off-by: joshrosso <rossoj@vmware.com>

joshrosso reviewed (changes requested):
fixed formatting; blocking to get feedback from engineering team.

joshrosso commented on ROADMAP.md:
```diff
@@ -2,60 +2,47 @@
-or in [community meetings, our Google Group, or our Slack channel](https://tanzucommunityedition.io/community/).
-Please open and comment on an issue if you want to provide suggestions and feedback to an item in the roadmap.
-Please review the roadmap to avoid potential duplicated effort.
+.Awaiting Proposal - these enhancements are in the idea mode, and the project maintainers are awaiting a detailed proposal. This state is useful to explore different solutions to a known problem and encourage discussion.
```
What is this `.` notation? Also these are are being formatted as a single paragraph. Did you mean to make this a bulleted list?

You can check your formatting by viewing the file in GitHub.

![image](https://user-images.githubusercontent.com/6200057/151604359-251f07db-856a-4239-9831-211094d74096.png)

qnetter replied:
Yes, it came from a "how to MD" doc but apparently an incorrect one...

joshrosso replied:
Looks great now, thanks for resolving this.

joshrosso commented on ROADMAP.md:
```diff
@@ -2,60 +2,47 @@
 
-More detail about short-term delivery plans can be found in the [milestones](https://github.com/vmware-tanzu/community-edition/milestones) in
-the [GitHub project repo](https://github.com/vmware-tanzu/community-edition).
+This roadmap should serve as a reference point for Tanzu Community Edition users to help us prioritize existing features and provide input on unmet needs, and for contributors to understand where the project is heading. A roadmap also helps our contributor community to determine if a proposed enhancement conflicts with the long-term plan.
```
>  A roadmap also helps our contributor community to determine if a proposed enhancement conflicts with the long-term plan.

I can't make sense of what this means.

qnetter replied:
If we said we plan to do X, if a contributor is thinking they want to contribute opposite-of-X, the roadmap will tell them it is un likely to be accepted.

joshrosso replied:
New working of:

```
Contributors can also consult the roadmap when thinking of proposing new ideas, to determine if their idea conflicts with the roadmap; in that case, the team and community will need to determine whether to adjust the roadmap or to recommend changes to the idea.
```

seems far more clear. Thanks for resolving this!

joshrosso commented on ROADMAP.md:
```diff
@@ -2,60 +2,47 @@
+This roadmap should serve as a reference point for Tanzu Community Edition users to help us prioritize existing features and provide input on unmet needs, and for contributors to understand where the project is heading. A roadmap also helps our contributor community to determine if a proposed enhancement conflicts with the long-term plan.
 
-## How to help
+Enhancement states help everyone understand if something is in the early stages of an idea or if it has wider community buy-in and committed resources. This is important because it helps limit waste, and focus work on the most important and timely activities. The stages are described below:
```
Recommend rewording this paragraph, it reads really weird.

Also, not sure there should be a command between `limit waste` and `and focus work`?

jpmcb replied:
Agreed ^ I'm not sure what `Enhancement states` are

joshrosso commented on ROADMAP.md:
```diff
@@ -2,60 +2,47 @@
-Check out our [issues](https://github.com/vmware-tanzu/community-edition/issues) for issues tagged
-[help-wanted](https://github.com/vmware-tanzu/community-edition/issues?q=is%3Aopen+is%3Aissue+label%3Ahelp-wanted) or
-[good-first-issue](https://github.com/vmware-tanzu/community-edition/labels/good-first-issue).
+Discussion on the roadmap can take place in threads under Issues or in community meetings, our Google Group, or our Slack channel. Please open and comment on an issue if you want to provide suggestions for feedback on  an item in the roadmap.  Community members are encouraged to be actively involved, and also stay informed so contributions can be made with the most positive effect and limited dup ...
```
```suggestion
Discussion on the roadmap can take place in threads under Issues or in community meetings, our Google Group, or our Slack channel. Please open and comment on an issue if you want to provide suggestions for feedback on an item in the roadmap.  Community members are encouraged to be actively involved, and also stay informed so contributions can be made with the most positive effect and limited duplication of effort.
```

qnetter replied:
Edits made manually.

joshrosso commented on ROADMAP.md:
```diff
@@ -2,60 +2,47 @@
-Check out our [issues](https://github.com/vmware-tanzu/community-edition/issues) for issues tagged
-[help-wanted](https://github.com/vmware-tanzu/community-edition/issues?q=is%3Aopen+is%3Aissue+label%3Ahelp-wanted) or
-[good-first-issue](https://github.com/vmware-tanzu/community-edition/labels/good-first-issue).
+Discussion on the roadmap can take place in threads under Issues or in community meetings, our Google Group, or our Slack channel. Please open and comment on an issue if you want to provide suggestions for feedback on  an item in the roadmap.  Community members are encouraged to be actively involved, and also stay informed so contributions can be made with the most positive effect and limited dup ...
```
`and also` reads weird.

qnetter replied:
Rewritten.

joshrosso commented on ROADMAP.md:
```diff
@@ -2,60 +2,47 @@
-Check out our [issues](https://github.com/vmware-tanzu/community-edition/issues) for issues tagged
-[help-wanted](https://github.com/vmware-tanzu/community-edition/issues?q=is%3Aopen+is%3Aissue+label%3Ahelp-wanted) or
-[good-first-issue](https://github.com/vmware-tanzu/community-edition/labels/good-first-issue).
+Discussion on the roadmap can take place in threads under Issues or in community meetings, our Google Group, or our Slack channel. Please open and comment on an issue if you want to provide suggestions for feedback on  an item in the roadmap.  Community members are encouraged to be actively involved, and also stay informed so contributions can be made with the most positive effect and limited dup ...
```
I also recommend re-working this paragraph. It doesn't read well (to me)

qnetter replied:
Rerwritten.

joshrosso commented on ROADMAP.md:
```diff
@@ -2,60 +2,47 @@
 
-Please open an issue to track any initiative on the roadmap (usually driven by new feature requests). We will work with
-and rely on our community to focus our efforts to improve Tanzu Community Edition.
+Please open an issue to track any initiative on the roadmap (usually driven by new feature requests). We will work with and rely on our community to focus our efforts to improve Tanzu Community Edition.
```
I don't understand this. Is it about "requesting a new item to be on the roadmap"? Like the title says?\

It says: "open an issue **to track any initiative on the roadmap**" <-- implies it pre-exists, right?

joshrosso commented on ROADMAP.md:
```diff
@@ -2,60 +2,47 @@
+| | Tools and processes for package contribution, maintenance, and installation; Package maintainer documentation and guidance is live such that everyone can understand the responsibilities of being a package maintainer | Awaiting Proposal | TBD |
+| |Provide guidance on bringing new provider packages to TCE; enabling the inclusion of new infrastructure providers | Awaiting Proposal (dependent on Tanzu Kubernetes Provider implementation) | TBD |
+| |Introduce kubeapps to TCE for discovering and installation packages in Bitnami library [(issue)](https://github.com/vmware-tanzu/community-edition/issues/2418) | Awaiting Proposal | TBD |
+| Better defined LCM Cluster API extensibility points |Build the Runtime Extension SDK in Cluster API for ClusterClass patching, pre/post control plane upgrade hooks | Awaiting proposal | TBD |
```
Trying to understand how this is part of the TCE roadmap.

Is the PM suggestion that the TCE team works on this upstream? Specifically we want the TCE project to "Build the Runtime Extension SDK in Cluster API for ClusterClass patching" ?

stmcginnis replied:
LCM should also be spelled out.

qnetter replied:
Moved to Exploration out of table.

joshrosso commented on ROADMAP.md:
```diff
@@ -2,60 +2,47 @@
-|**Tanzu scenario support beyond Kubernetes**|Tanzu Application Platform OSS components|0-3 months|
-||Add developer focused UI experience|0-3 months|
-||Bitnami integration via kubeapps|0-3 months|
+The following table includes the current roadmap for Tanzu Community Edition. If you have any questions or would like to contribute, please attend a community meeting to discuss with our team. If you don't know where to start, we are always looking for contributors that will help us reduce technical, automation, and documentation debt. Please take the timelines and dates as proposals and goals. P ...
```
These paragraph feels like a re-hash of a bunch of detail covered throughout this doc already.

jpmcb replied:
^ +1

joshrosso commented on ROADMAP.md:
```diff
@@ -2,60 +2,47 @@
-||Bitnami integration via kubeapps|0-3 months|
+The following table includes the current roadmap for Tanzu Community Edition. If you have any questions or would like to contribute, please attend a community meeting to discuss with our team. If you don't know where to start, we are always looking for contributors that will help us reduce technical, automation, and documentation debt. Please take the timelines and dates as proposals and goals. P ...
+
+Last Updated:  January 2022
```
Please update to Feb.

tbh, not sure why this is needed considering there is commit history, but if we keep it, please update to feb.

joshrosso commented on ROADMAP.md:
```diff
@@ -2,60 +2,47 @@
+
+| Theme | Feature | Status (Phase) | Targeted Release |
+| ----- | ------- | -------------- | ---------------- |
+| Build user community | Reward community support | n/a | ongoing |
```
:-1: to this as a roadmap item.

it doesn't benefit users of the project, it's not measurable, and feels like either an internal detail or random filler.

dvonthenen replied:
Agreed. This doesnt pertain to the work/tasks needing to be done in this repo. This should be a goal on an internal tracker like something for community management.

jpmcb replied:
^ +1 - Building a user community is inherent in a community driven project; I'm not sure how you measure, define success, or re-evaluate failure.

qnetter replied:
I've talked it over with others too - let's remove it and be sure we track it in the program.

qnetter replied:
Removed.

joshrosso commented on ROADMAP.md:
```diff
@@ -2,60 +2,47 @@
+| Theme | Feature | Status (Phase) | Targeted Release |
+| ----- | ------- | -------------- | ---------------- |
+| Build user community | Reward community support | n/a | ongoing |
+| |Community roadmap prioritization | n/a | ongoing |
```
This is a roadmap item? Isn't this what the roadmap is?

qnetter replied:
Nigel's activities. (In this case, the activity is to hand out swag for attendance and contribution.) Marketing activities. Follow-up to conversations that draw people (obviously not too effectively so far) to the channels and meetings. At the program level, deliverables are not all code and docs.

stmcginnis replied:
Said another way, it doesn't make sense for a user to read the roadmap to see what's coming up and think something like "great, I can't wait for the v0.12.0 release when we get a community". This just seems like a very odd thing to have on a roadmap. A personal or team goal planning, sure. But a roadmap not so much.

qnetter replied:
Yes, I get your point. In the previous version, this was one of several items in this category, so "Reward participation" as an activity made more sense in context. It also included several dev advocacy and technical marketing activities aimed at building the community. The content plan has not yet been approved by the program - it has, I think, by marketing - so I was asked to drop those items. I could drop the whole category till those things are approved, but since the (potential) community is one of the things we talk about a lot when we present TCE to Tanzu customers, press, analysts, etc. the general category should probably remain - we can revise to add the other deliverables as soon as the plan is approved. But I will discuss it with People Above My Head and see what they think.

qnetter replied:
Removed.

joshrosso commented on ROADMAP.md:
```diff
@@ -2,60 +2,47 @@
+| ----- | ------- | -------------- | ---------------- |
+| Build user community | Reward community support | n/a | ongoing |
+| |Community roadmap prioritization | n/a | ongoing |
+| Enhance cloud-native user experience for skilled and new users | Unmanaged-cluster model introduced to enable minimal-cluster deployments in under 4 minutes that can run on consumer hardware. | Implementation in Progress | v0.11 |
```
:+1: really love this one. (just positive feedback, feel free to close)

stmcginnis replied:
Do we want it to say v0.11? The current plan is to include it in the v0.10.0 release, correct?

jpmcb replied:
Un-resolved this one since it still looks like it's marked for v0.11?

qnetter replied:
Unmanaged now says 0.10.

joshrosso commented on ROADMAP.md:
```diff
@@ -2,60 +2,47 @@
+| | Work to reduce time to cluster availability on vSphere and public clouds (Cluster API v1.1 Proposal) | Awaiting Proposal | TBD |
+| | Make management clusters registerable in Tanzu Mission Control | Implementation in progress | v0.11 |
+| | Introduce a UI that takes a user from installation through creating clusters to installing packages on clusters. Embed documentation and guidance throughout so that users can be successful exclusively through this tooling. | Awaiting Proposal | TBD |
+| | Expose Cluster API (capi)-provider bootstrapping details, client-side, to users when creating management and workload clusters. Provides users an understanding of where bootstrapping failures occurred. | Awaiting Proposal [(issue)](https://github.com/vmware-tanzu/community-edition/issues/2730 ) | TBD |
```
+1 really love this one. (just positive feedback, feel free to close/resolve)

joshrosso commented on ROADMAP.md:
```diff
@@ -2,60 +2,47 @@
+| Enhance cloud-native user experience for skilled and new users | Unmanaged-cluster model introduced to enable minimal-cluster deployments in under 4 minutes that can run on consumer hardware. | Implementation in Progress | v0.11 |
+| | Work to reduce time to cluster availability on vSphere and public clouds (Cluster API v1.1 Proposal) | Awaiting Proposal | TBD |
+| | Make management clusters registerable in Tanzu Mission Control | Implementation in progress | v0.11 |
+| | Introduce a UI that takes a user from installation through creating clusters to installing packages on clusters. Embed documentation and guidance throughout so that users can be successful exclusively through this tooling. | Awaiting Proposal | TBD |
```
+1 really love this one. (just positive feedback, feel free to close/resolve)

joshrosso commented on ROADMAP.md:
```diff
@@ -2,60 +2,47 @@
+| |Community roadmap prioritization | n/a | ongoing |
+| Enhance cloud-native user experience for skilled and new users | Unmanaged-cluster model introduced to enable minimal-cluster deployments in under 4 minutes that can run on consumer hardware. | Implementation in Progress | v0.11 |
+| | Work to reduce time to cluster availability on vSphere and public clouds (Cluster API v1.1 Proposal) | Awaiting Proposal | TBD |
+| | Make management clusters registerable in Tanzu Mission Control | Implementation in progress | v0.11 |
```
From what i've read of this doc, it feels like Implementation in progress implies an issue exists and proposal is accepted. I haven't seen this?

jpmcb replied:
> management clusters registerable

Is this work that the TCE team can execute against? The `management-cluster` plugin and core functionality for management clusters is owned by `tanzu-framework`

joshrosso commented on ROADMAP.md:
```diff
@@ -2,60 +2,47 @@
+| |Provide guidance on bringing new provider packages to TCE; enabling the inclusion of new infrastructure providers | Awaiting Proposal (dependent on Tanzu Kubernetes Provider implementation) | TBD |
+| |Introduce kubeapps to TCE for discovering and installation packages in Bitnami library [(issue)](https://github.com/vmware-tanzu/community-edition/issues/2418) | Awaiting Proposal | TBD |
+| Better defined LCM Cluster API extensibility points |Build the Runtime Extension SDK in Cluster API for ClusterClass patching, pre/post control plane upgrade hooks | Awaiting proposal | TBD |
+| Accelerate time to bootstrap Cluster API management clusters | Re-evaluate the Cluster API bootstrapping process to use `kcp-dev` (idea) | Awaiting proposal | TBD |
```
In the upper part of the ROADMAP doc, it implies `Awaiting Proposal` is still in ideation. Do we need to say this is an "(idea), here?

joshrosso commented on ROADMAP.md:
```diff
@@ -2,60 +2,47 @@
+| | Validate and improve end-to-end tests across all supported infrastructure providers. | Awaiting proposal | TBD |
+| Rich package library | Work with community to solicit and prioritize desired packages| n/a |ongoing |
+| | Pursue community and ISV package lifecycle and configuration integrations via Carvel and Tanzu Framework | n/a | ongoing |
+| Positioning Community rdition as leading-edge upstream for Tanzu products | Community Edition produces its own Bill of Materials (BOM), Tanzu Kubernetes Release (TKr), OCI container images, host images, and packages for CLI plugins and in-cluster software | Awaiting Proposals | TBD |
```
```suggestion
| Positioning Community Edition as leading-edge upstream for Tanzu products | Community Edition produces its own Bill of Materials (BOM), Tanzu Kubernetes Release (TKr), OCI container images, host images, and packages for CLI plugins and in-cluster software | Awaiting Proposals | TBD |
```

joshrosso reviewed (changes requested):
my general feedback is this roadmap needs a lot of love.

I find it hard to read and understand.

It has issues ranging from formatting to content.

qnetter committed:
Update ROADMAP.md

Co-authored-by: joshrosso <rossoj@vmware.com>

qnetter commented on ROADMAP.md:
```diff
@@ -2,60 +2,47 @@
+| |Provide guidance on bringing new provider packages to TCE; enabling the inclusion of new infrastructure providers | Awaiting Proposal (dependent on Tanzu Kubernetes Provider implementation) | TBD |
+| |Introduce kubeapps to TCE for discovering and installation packages in Bitnami library [(issue)](https://github.com/vmware-tanzu/community-edition/issues/2418) | Awaiting Proposal | TBD |
+| Better defined LCM Cluster API extensibility points |Build the Runtime Extension SDK in Cluster API for ClusterClass patching, pre/post control plane upgrade hooks | Awaiting proposal | TBD |
+| Accelerate time to bootstrap Cluster API management clusters | Re-evaluate the Cluster API bootstrapping process to use `kcp-dev` (idea) | Awaiting proposal | TBD |
```
That's great - I didn't write it in - I thought you did, on the Google Doc.

joshrosso replied:
I did not write this in. Thanks for changing the wording.

qnetter committed:
Update ROADMAP.md

kartiklunkad26 commented:
Could I know more about the "developer UI experience" roadmap item?

joshrosso commented on ROADMAP.md:
```diff
@@ -2,60 +2,47 @@
 
-More detail about short-term delivery plans can be found in the [milestones](https://github.com/vmware-tanzu/community-edition/milestones) in
-the [GitHub project repo](https://github.com/vmware-tanzu/community-edition).
+This roadmap should serve as a reference point for Tanzu Community Edition users to help us prioritize existing features and provide input on unmet needs, and for contributors to understand where the project is heading. Contributors can also consult the roadmap when thinking of proposing new ideas, to determine if their idea conflicts with the roadmap; in that case, the team and community will ne ...
```
```suggestion
This roadmap should serve as a reference point for Tanzu Community Edition users to help us prioritize existing features, provide input on unmet needs, and for contributors to understand where the project is heading. Contributors can also consult the roadmap when thinking of proposing new ideas and determine if their idea conflicts with the roadmap. If it does conflict, the team and community will need to determine whether to adjust the roadmap or to recommend changes to the idea.
```

joshrosso commented:
> Could I know more about the "developer UI experience" roadmap item?

@kartiklunkad26, you may be looking at an old commit?

As I understand PM's plan, TCE has no plans of targeting a developer-focused UI.

Here is the item that relates to a UI:

![image](https://user-images.githubusercontent.com/6200057/151826588-a88c0bd2-3a97-4af5-a90a-bb84430cf124.png)

If you want details on that UI idea, @qnetter can fill you in.

dvonthenen commented on ROADMAP.md:
```diff
@@ -2,60 +2,47 @@
+| | Validate and improve end-to-end tests across all supported infrastructure providers. | Awaiting proposal | TBD |
+| Rich package library | Work with community to solicit and prioritize desired packages| n/a |ongoing |
+| | Pursue community and ISV package lifecycle and configuration integrations via Carvel and Tanzu Framework | n/a | ongoing |
+| Positioning Community Edition as leading-edge upstream for Tanzu products | Community Edition produces its own Bill of Materials (BOM), Tanzu Kubernetes Release (TKr), OCI container images, host images, and packages for CLI plugins and in-cluster software | Awaiting Proposals | TBD |
```
Currently, I think plans are being developed in Tanzu Framework and by said team. If that happens to be the case, I think we should call out that we are consumers of this feature and we should link to the proposal and again note the work isn't done in this repo.

qnetter replied:
@dvonthenen please Slack me when you get this comment. :)

davidvonthenen reviewed (commented):
Left some comments and looking for clarification on a couple of things

qnetter commented:
@dvonthenen This is not a roadmap for the repo. A roadmap for the repo is useful to contributors but less so to users. Users need to see a roadmap for the entire project, not just for the repo.

seemiller reviewed (dismissed):
Package related tasks seem reasonable.

garrying commented on ROADMAP.md:
```diff
@@ -2,60 +2,47 @@
+| | Pursue community and ISV package lifecycle and configuration integrations via Carvel and Tanzu Framework | n/a | ongoing |
+| Positioning Community Edition as leading-edge upstream for Tanzu products | Community Edition produces its own Bill of Materials (BOM), Tanzu Kubernetes Release (TKr), OCI container images, host images, and packages for CLI plugins and in-cluster software | Awaiting Proposals | TBD |
+| Application Platform | Installable package is made available containing pre-configured, minimally-viable, platform services that enable developers to run their tools and apps (contour, kpack, cert-manager, knative serving, Cartographer, etc.) | Implementation in Progress (TODO: tracker proposal for all components) | Beginning in v0.10; please see issue (in dev) |
+| | Tools and processes for package contribution, maintenance, and installation; Package maintainer documentation and guidance is live such that everyone can understand the responsibilities of being a package maintainer | Awaiting Proposal | TBD |
```
Nit: I think this could be two rows for `Tools and processes...` and `Package maintainer documentation...` for clarity.

qnetter replied:
Changed as noted.

jpmcb commented on ROADMAP.md:
```diff
@@ -2,60 +2,47 @@
-Check out our [issues](https://github.com/vmware-tanzu/community-edition/issues) for issues tagged
-[help-wanted](https://github.com/vmware-tanzu/community-edition/issues?q=is%3Aopen+is%3Aissue+label%3Ahelp-wanted) or
-[good-first-issue](https://github.com/vmware-tanzu/community-edition/labels/good-first-issue).
+Discussion on the roadmap can take place in threads under Issues or in community meetings, our Google Group, or our Slack channel. Please open and comment on an issue if you want to provide suggestions for feedback on  an item in the roadmap.  Community members are encouraged to be actively involved, and also stay informed so contributions can be made with the most positive effect and limited dup ...
```
> Discussion on the roadmap can take place in threads under Issues or in community meetings, our Google Group, or our Slack channel

This seems really ambiguous to me. If I'm a new contributor, or even an existing contributor, it's overwhelming to be expected to track and communicate within so many different avenues without clear direction.

Should we make a recommendation on the _preferred_ communication channel?

i.e.

```diff
- Discussion on the roadmap can take place in threads under Issues or in community meetings, our Google Group, or our Slack channel
+ Discussion on the roadmap should take place in threads under Issues. Although we welcome discussion in community meetings, our Google Group, or our Slack channel.
```

qnetter replied:
Updated.

jpmcb commented on ROADMAP.md:
```diff
@@ -2,60 +2,47 @@
+| Application Platform | Installable package is made available containing pre-configured, minimally-viable, platform services that enable developers to run their tools and apps (contour, kpack, cert-manager, knative serving, Cartographer, etc.) | Implementation in Progress (TODO: tracker proposal for all components) | Beginning in v0.10; please see issue (in dev) |
+| | Tools and processes for package contribution, maintenance, and installation; Package maintainer documentation and guidance is live such that everyone can understand the responsibilities of being a package maintainer | Awaiting Proposal | TBD |
+| |Provide guidance on bringing new provider packages to TCE; enabling the inclusion of new infrastructure providers | Awaiting Proposal (dependent on Tanzu Kubernetes Provider implementation) | TBD |
+| |Introduce kubeapps to TCE for discovering and installation packages in Bitnami library [(issue)](https://github.com/vmware-tanzu/community-edition/issues/2418) | Awaiting Proposal | TBD |
```
Doesn't this conflict with this goal?

> Pursue community and ISV package lifecycle and configuration integrations via Carvel and Tanzu Framework

If we introduced the kubeapps bitnami library, wouldn't that fracture our package eco-system and make installing / maintaining packages _more_ challenging for users and maintainers?

jpmcb commented on ROADMAP.md:
```diff
@@ -2,60 +2,47 @@
+| |Provide guidance on bringing new provider packages to TCE; enabling the inclusion of new infrastructure providers | Awaiting Proposal (dependent on Tanzu Kubernetes Provider implementation) | TBD |
+| |Introduce kubeapps to TCE for discovering and installation packages in Bitnami library [(issue)](https://github.com/vmware-tanzu/community-edition/issues/2418) | Awaiting Proposal | TBD |
+| Better defined LCM Cluster API extensibility points |Build the Runtime Extension SDK in Cluster API for ClusterClass patching, pre/post control plane upgrade hooks | Awaiting proposal | TBD |
+| Accelerate time to bootstrap Cluster API management clusters | Consider more efficient and resource-saving methods to bootstrap | Awaiting proposal | TBD |
```
I don't understand this one.

Are we saying that the TCE team should execute against the `Cluster-API` library (aka, make upstream contributions) in order to make the bootstrapping of a Cluster API management cluster faster?

Or is this work that `tanzu-framework` needs to execute against in order to speed up the process of bootstrapping a management cluster? (this makes the most sense to me since they are pulling in that library and harnessing the bootstraping capabilities of Cluster API)

Or is this saying that we _completely reconsider_ using Cluster API in order to find "more efficient and resource-saving methods to bootstrap"?

qnetter commented:
Any way to respond to individual comments?

qnetter commented:
> > Any way to respond to individual comments?
> 
> Yes, each comment has a box at the bottom of it where you can respond to the specific comment.
> 
> ![image](https://user-images.githubusercontent.com/6200057/151857237-cffb3c3b-3c03-49d4-9b5e-65a4fd23f7d6.png)
> 
> Hope this helps.

Misspoke - I mean REVIEW comments. I am somehow not seeing that there.

joshrosso commented:
> > > Any way to respond to individual comments?
> > 
> > 
> > Yes, each comment has a box at the bottom of it where you can respond to the specific comment.
> > ![image](https://user-images.githubusercontent.com/6200057/151857237-cffb3c3b-3c03-49d4-9b5e-65a4fd23f7d6.png)
> > Hope this helps.
> 
> Misspoke - I mean REVIEW comments. I am somehow not seeing that there.

Those are review comments :thinking: ?

Feel free to hit me up on slack/zoom and we can take a look at what you're seeing.

qnetter committed:
Update ROADMAP.md

Explained that the roadmap is for the whole program including other repos and non-code, not just for the community-edition repo.

qnetter committed:
Update ROADMAP.md

Clarify roadmap purpose for users ands contributors.

Co-authored-by: joshrosso <rossoj@vmware.com>

stmcginnis commented on ROADMAP.md:
```diff
@@ -2,60 +2,49 @@
+| Application Platform | Installable package is made available containing pre-configured, minimally-viable, platform services that enable developers to run their tools and apps (contour, kpack, cert-manager, knative serving, Cartographer, etc.) | Implementation in Progress (TODO: tracker proposal for all components) | Beginning in v0.10; please see issue (in dev) |
+| | Tools and processes for package contribution, maintenance, and installation; Package maintainer documentation and guidance is live such that everyone can understand the responsibilities of being a package maintainer | Awaiting Proposal | TBD |
+| |Provide guidance on bringing new provider packages to TCE; enabling the inclusion of new infrastructure providers | Awaiting Proposal (dependent on Tanzu Kubernetes Provider implementation) | TBD |
+| |Introduce kubeapps to TCE for discovering and installation packages in Bitnami library [(issue)](https://github.com/vmware-tanzu/community-edition/issues/2418) | Awaiting Proposal | TBD |
```
"for discovering and installation packages in Bitnami library" should be "for discovering and installing packages from the Bitnami library" or "for the discovery and installation of packages from the Bitnami library".

qnetter committed:
Update ROADMAP.md

Spell out LCM.

nrb commented on ROADMAP.md:
```diff
@@ -2,60 +2,49 @@
+| |Provide guidance on bringing new provider packages to TCE; enabling the inclusion of new infrastructure providers | Awaiting Proposal (dependent on Tanzu Kubernetes Provider implementation) | TBD |
+| |Introduce kubeapps to TCE for discovering and installation packages in Bitnami library [(issue)](https://github.com/vmware-tanzu/community-edition/issues/2418) | Awaiting Proposal | TBD |
+| Better defined lifecycle management Cluster API extensibility points |Build the Runtime Extension SDK in Cluster API for ClusterClass patching, pre/post control plane upgrade hooks | Awaiting proposal | TBD |
+| Accelerate time to bootstrap Cluster API management clusters | Consider more efficient and resource-saving methods to bootstrap | Awaiting proposal | TBD |
```
I'm not following why this theme is separate from `Enhance cloud-native user experience for skilled and new users`, which includes a faster setup time for unmanaged clusters.

In my mind, these are similar or the same, but perhaps I'm missing something.

nrb commented on ROADMAP.md:
```diff
@@ -2,60 +2,49 @@
 
-More detail about short-term delivery plans can be found in the [milestones](https://github.com/vmware-tanzu/community-edition/milestones) in
-the [GitHub project repo](https://github.com/vmware-tanzu/community-edition).
+The scope of this roadmap is the entire Tanzu Community Edition program, which includes this GitHub project repository, the [Tanzu Framework](https://github.com/vmware-tanzu/tanzu-framework) project frepository, the project repositories for packages delivered as part of Tanzu Comm unity Edition, and non-code program aspects such as videos, training materials, and community activities.
```
> The scope of this roadmap is the entire Tanzu Community Edition program, which includes this GitHub project repository, the **Tanzu Framework project repository**

Emphasis mine. Does this mean we are claiming and owning the roadmap for Framework?

qnetter committed:
Update ROADMAP.md

Merge bootstrap acceleration items.

qnetter committed:
Update ROADMAP.md

Reworded roadmap discussion guidelines to prioritize Issues and scheduled community meeting sessions.

qnetter committed:
Update ROADMAP.md

Split package maintainer item into two

qnetter committed:
Update ROADMAP.md

Lots of rewording to reduce duplication, resurface good first issues, etc.

qnetter committed:
Update ROADMAP.md

Fixed the date.

qnetter committed:
Update ROADMAP.md

Updated release of unmanaged-cluster to 0.10

qnetter committed:
Update ROADMAP.md

Fixed awkward Bitnami/kubeapps item.

qnetter commented:
> > Could I know more about the "developer UI experience" roadmap item?
> 
> @kartiklunkad26, you may be looking at an old commit?
> 
> As I understand PM's plan, TCE has no plans of targeting a developer-focused UI.
> 
> Here is the item that relates to a UI:
> 
> ![image](https://user-images.githubusercontent.com/6200057/151826588-a88c0bd2-3a97-4af5-a90a-bb84430cf124.png)
> 
> If you want details on that UI idea, @qnetter can fill you in.

At some point, the Tanzu Application Platform OSS components in Tanzu Community Edition will include developer dashboards.

qnetter commented:
@vmware-tanzu/tce-owners please re-review.

qnetter committed:
Update ROADMAP.md

mdlint fix.

qnetter committed:
Update ROADMAP.md

This time mdlint.

dvonthenen commented:
> @dvonthenen This is not a roadmap for the repo. A roadmap for the repo is useful to contributors but less so to users. Users need to see a roadmap for the entire project, not just for the repo.

I'm totally cool with that... my only comment was if the work isn't being done in the repo, we should point to where the work and/or proposals are being done (ie pointers to issues and proposals in other repos aka Tanzu Framework).

jpmcb commented on ROADMAP.md:
```diff
@@ -1,61 +1,55 @@
+| | Tools and processes for package contribution, maintenance, and installation | Awaiting Proposal | TBD |
+| | Package maintainer documentation and guidance is live such that everyone can understand the responsibilities of being a package maintainer | Awaiting Proposal | TBD |
+| |Provide guidance on bringing new provider packages to TCE; enabling the inclusion of new infrastructure providers | Awaiting Proposal (dependent on Tanzu Kubernetes Provider implementation) | TBD |
+| |Introduce kubeapps to TCE for discovery and installation of packages from Bitnami library [(issue)](https://github.com/vmware-tanzu/community-edition/issues/2418) | Awaiting Proposal | TBD |
```
Thought I had made this comment earlier, but this bitnami goal seems to be in opposition to this goal:

> Pursue community and ISV package lifecycle and configuration integrations via Carvel and Tanzu Framework

Won't introducing another packaging paradigm create fracture the packaging community and make it _more_ difficult to get started easily with packages and a full running platform? Yes, we have many existing assets in the Bitnami library, but in my eyes, it's risky introducing them as an _alternative_ to our established packaging philosophy.

Further, does this create the expectation that _maintainers_ of packages maintain _both_ their carvel package _and_ the bitmani chart? If I'm the packaging dev for prometheus, this seems like extra work when I've potentially already created automation, tooling, and workflows for the carvel packaging set.

I'd love @seemiller's perspective on this too.

qnetter replied:
There are many packages in Bitnami that are not in our repos. And they are duplicating packages in Carvel tooling.  At this point there is very little overlap, and we should advise users to search the TCE repo before Kubeapps.

qnetter commented on ROADMAP.md:
```diff
@@ -1,61 +1,55 @@
 
-More detail about short-term delivery plans can be found in the [milestones](https://github.com/vmware-tanzu/community-edition/milestones) in
-the [GitHub project repo](https://github.com/vmware-tanzu/community-edition).
+The scope of this roadmap is the entire Tanzu Community Edition program, which includes this GitHub project repository, the [Tanzu Framework](https://github.com/vmware-tanzu/tanzu-framework) project frepository, the project repositories for packages delivered as part of Tanzu Comm unity Edition, and non-code program aspects such as videos, training materials, and community activities.
```
Fixed.

qnetter reviewed (commented):
I guess I didn't realize I had to submit before comments could be seen. Sorry that my ignorance caused worry and upset..

qnetter committed:
Update ROADMAP.md

Typos.

qnetter committed:
Update ROADMAP.md

Moved CAPI issues to Exploration section and out of table

qnetter commented on ROADMAP.md:
```diff
@@ -2,60 +2,47 @@
+| Build user community | Reward community support | n/a | ongoing |
+| |Community roadmap prioritization | n/a | ongoing |
+| Enhance cloud-native user experience for skilled and new users | Unmanaged-cluster model introduced to enable minimal-cluster deployments in under 4 minutes that can run on consumer hardware. | Implementation in Progress | v0.11 |
+| | Work to reduce time to cluster availability on vSphere and public clouds (Cluster API v1.1 Proposal) | Awaiting Proposal | TBD |
```
I have moved this item to an Other Areas To Explore section below the table.

qnetter reviewed (commented):
Hope this goes and the comments were all sent.

qnetter reviewed (commented):
I hope comments were received.

@vmware-tanzu/tce-owners Please re-review.

stmcginnis reviewed (dismissed):
My concerns have been addressed.

kcoriordan commented on ROADMAP.md:
```diff
@@ -1,61 +1,57 @@
 [help-wanted](https://github.com/vmware-tanzu/community-edition/issues?q=is%3Aopen+is%3Aissue+label%3Ahelp-wanted) or
 [good-first-issue](https://github.com/vmware-tanzu/community-edition/labels/good-first-issue).
 
+You can participate in discussion on the roadmap in threads under Issues and in sdcheduled roadmap serssions in community meetings. Questions may also be brought up in our Google Group or our Slack channel. Community members are encouraged to be actively involved, and also stay informed so contributions can be made with the most positive effect and limited duplication of effort.
```
```suggestion
You can participate in discussion on the roadmap in threads under Issues and in scheduled roadmap sessions in community meetings. Questions may also be brought up in our Google Group or our Slack channel. Community members are encouraged to be actively involved, and also stay informed so contributions can be made with the most positive effect and limited duplication of effort.
```

qnetter committed:
Update ROADMAP.md

typos. Thanks @kcoriordan !

Co-authored-by: Kay ORiordan <koriordan@vmware.com>

joshrosso reviewed (approved):
We've decided to merge this and iterate on future suggestions through github issues.

Thanks everyone!
